src/config.ts

```diff
@@ -1,16 +1,55 @@
 import { config } from 'dotenv';
+import { ethers } from 'ethers';
 
 // Load environment variables
 config();
 
-if (!process.env.SUPABASE_URL || !process.env.SUPABASE_KEY) {
-  throw new Error('Missing Supabase credentials in environment variables');
+// Validate required environment variables
+const requiredEnvVars = [
+  'RPC_URL',
+  'STAKER_CONTRACT_ADDRESS',
+  'SUPABASE_URL',
+  'SUPABASE_KEY',
+] as const;
+
+for (const envVar of requiredEnvVars) {
+  if (!process.env[envVar]) {
+    throw new Error(`Missing required environment variable: ${envVar}`);
+  }
 }
 
 export const CONFIG = {
   supabase: {
     url: process.env.SUPABASE_URL,
     key: process.env.SUPABASE_KEY,
-    db: process.env.DB,
+  },
+  monitor: {
+    rpcUrl: process.env.RPC_URL!,
+    chainId: parseInt(process.env.CHAIN_ID || '42161'),
```

Comment:
I would make this a required field rather than providing a default.

Reply:
done